homeassistant/components/wirelesstag/__init__.py

```diff
@@ -56,15 +58,36 @@
 )
 
 
-class WirelessTagPlatform:
+class WirelessTagPlatform(HomeAssistantView):
     """Principal object to manage all registered in HA tags."""
 
+    requires_auth = False
+    url = "/api/wirelesstag/{event}"
```

Comment:
I think that we should require a password to be added, or else anyone can send data to any HA instance.

Reply:
The problem is that the tagmanager cannot use any real form of authentication (no forms, no digist, no basic). It can pass something on the querystring, but aside from that, it cannot do anything else. (I'm not even sure it can post to SSL)

I thought about doing this:
https://github.com/home-assistant/core/pull/36412#issuecomment-638454455

Which would basically register a randomized endpoint. Again, I cannot use a full guid, but 16 characters, random every time HA is started would be much better than nothing.

I have that big of code running on my machine right now. I would include it in the PR if that is something you'd allow.

The 148 character limit is very limiting. People have that an internal vanity url that is larger than an ip address is really chewing into that 148 character limit.

Reply:
Adding the random key to the API url should be fine, even 6 random characters is better than nothing. `/api/wirelessstag/{key}/{event}`